src/main/scala/org/ergoplatform/local/TransactionGenerator.scala

```diff
@@ -81,6 +80,47 @@ class TransactionGenerator(viewHolder: ActorRef,
 
     case SuccessfulTransaction(_) => self ! Attempt
   }
+
+  private def genTransaction(wallet: ErgoWallet): Future[Try[ErgoTransaction]] = {
+    val feeReq = PaymentRequest(Pay2SAddress(Values.TrueLeaf), fee, None, None)
+    val amountToPay = (Random.nextInt(10) + 1) * 100000000
+    val paymentReq = PaymentRequest(randProposition, amountToPay, None, None)
+    val tokenPaymentReq = wallet.confirmedBalances().map { balances =>
+      if (balances.assetBalances.nonEmpty) {
+        val tokenToSpend = balances.assetBalances.toSeq(Random.nextInt(balances.assetBalances.size))
+        val tokenAmountToSpend = tokenToSpend._2 / 4
+        Algos.decode(tokenToSpend._1).map { id =>
+          PaymentRequest(randProposition, 0L, Some(Seq(Digest32 @@ id -> tokenAmountToSpend)), None)
+        }.toOption
+      } else { None }
+    }
+    val assetIssueReq = wallet.inputsFor(Seq(feeReq, paymentReq)).map { inputs =>
+      inputs.headOption.map { firstInput =>
+        val assetId = Digest32 !@@ firstInput.id
+        val assetInfo = genNewAssetInfo
+        AssetIssueRequest(randProposition, assetId, assetInfo._1, assetInfo._2,
+          assetInfo._3, assetInfo._4)
+      }
+    }
+    tokenPaymentReq.flatMap { tpOutOpt =>
+      assetIssueReq.flatMap { aiOutOpt =>
+        val requests = Seq(tpOutOpt, aiOutOpt).foldLeft[Seq[TransactionRequest]](Seq(feeReq, paymentReq)) {
```

Comment:
Might be rewritten easier, e.g. `Seq(tpOutOpt, aiOutOpt, Some(feeReq), Some(paymentReq)).flatten`